games-fps/yamagi-quake2/yamagi-quake2-8.00.ebuild

```diff
@@ -0,0 +1,151 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit desktop wrapper
+
+CTF_V="1.08"
+ROGUE_V="2.08"
+XATRIX_V="2.09"
+REF_VK_V="1.0.1"
+
+DESCRIPTION="Quake 2 engine focused on single player"
+HOMEPAGE="https://www.yamagi.org/quake2/"
+SRC_URI="https://deponie.yamagi.org/quake2/quake2-${PV}.tar.xz
+	ctf? ( https://deponie.yamagi.org/quake2/quake2-ctf-${CTF_V}.tar.xz )
+	rogue? ( https://deponie.yamagi.org/quake2/quake2-rogue-${ROGUE_V}.tar.xz )
+	xatrix? ( https://deponie.yamagi.org/quake2/quake2-xatrix-${XATRIX_V}.tar.xz )
+	vulkan? ( https://github.com/yquake2/ref_vk/archive/refs/tags/v${REF_VK_V}.tar.gz ->
+		quake2-ref_vk-${REF_VK_V}.tar.gz )
+"
+S="${WORKDIR}/quake2-${PV}"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~arm64 ~x86"
+IUSE="+client ctf dedicated openal +opengl rogue softrender vulkan xatrix"
+REQUIRED_USE="
+	|| ( client dedicated )
+	client? ( || ( opengl softrender vulkan ) )
+"
+
+COMMON_DEPEND="
+	client? (
+		media-libs/libsdl2[video]
+		!openal? ( media-libs/libsdl2[sound] )
+		opengl? (
+			media-libs/libsdl2[opengl]
+			virtual/opengl
+		)
+	)
+"
+DEPEND="${COMMON_DEPEND}
+	client? ( vulkan? ( dev-util/vulkan-headers ) )
+"
+RDEPEND="${COMMON_DEPEND}
+	client? ( openal? ( media-libs/openal ) )
+"
+
+DOCS=( CHANGELOG README.md doc )
+
+mymake() {
+	emake \
+		VERBOSE=1 \
+		WITH_SYSTEMWIDE=yes \
+		WITH_SYSTEMDIR="${EPREFIX}"/usr/share/quake2 \
+		WITH_OPENAL=$(usex openal) \
+		"$@"
+}
+
+src_compile() {
+	local targets=( game )
+	if use client; then
+		targets+=( client )
+		use opengl && targets+=( ref_gl1 ref_gl3 )
+		use softrender && targets+=( ref_soft )
+	fi
+	use dedicated && targets+=( server )
+
+	mymake config
+	mymake "${targets[@]}"
+
+	if use client && use vulkan; then
+		emake -C "${WORKDIR}"/ref_vk-${REF_VK_V} VERBOSE=1
+	fi
+
+	local addon
+	for addon in ctf rogue xatrix; do
+		use ${addon} || continue
+		emake -C "${WORKDIR}"/quake2-${addon}-* VERBOSE=1
+	done
+}
+
+src_install() {
+	insinto /usr/lib/yamagi-quake2
+	# Yamagi Quake II expects all binaries to be in the same directory
+	# See stuff/packaging.md for more info
+	exeinto /usr/lib/yamagi-quake2
+	doins -r release/.
+
+	if use client; then
+		doexe release/quake2
+		dosym ../lib/yamagi-quake2/quake2 /usr/bin/yquake2
+
+		newicon stuff/icon/Quake2.svg "yamagi-quake2.svg"
+		make_desktop_entry "yquake2" "Yamagi Quake II"
+	fi
+
+	if use client && use vulkan; then
+		insinto /usr/lib/yamagi-quake2
+		doins "${WORKDIR}"/ref_vk-${REF_VK_V}/release/ref_vk.so
+	fi
```

Comment:
Nevermind the last bit, I just realized how this handles loading renderers and takes any values without checking if it's right.

Reply:
> Maybe move this inside the `use client` block just above rather than check client again?

Then the vulkan target will be the first in the build log, even before `make config` output. Also I think this block should be logically separated.

Reply:
This bit was about the one in src_install, make config was in src_compile and unrelated.

But if want to keep them separate it's fine too.